tony-history-server/app/cache/CacheWrapper.java

```diff
@@ -24,9 +25,18 @@
    */
   private static Cache<String, List<JobConfig>> configCache;
 
+  /**
+   * eventCache
+   * - key: job ID (application_[0-9]+_[0-9]+)
+   * - value: List of JobEvent objects. Each JobEvent object
+   * represents an Event in job's jhist
+   */
+  private static Cache<String, List<JobEvent>> eventCache;
+
   public CacheWrapper() {
     metadataCache = CacheBuilder.newBuilder().build();
     configCache = CacheBuilder.newBuilder().build();
+    eventCache = CacheBuilder.newBuilder().build();
```

Comment:
Do these caches have a max size? Should we set a max size? What is their behavior when memory is scarce?

Reply:
Please specify a maximumSize when creating these caches. (This should be configurable in the tony-site.xml file.) Otherwise, we'll eventually run out of memory and the service will crash.